packages/react/src/components/toggle.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { clx } from "../utils";
+
+type ToggleSize = "medium" | "large";
+
+export interface ToggleProps {
+  label?: string;
+  defaultChecked?: boolean;
+  checked?: boolean;
+  onCheckedChange?: (checked: boolean) => void;
+  size?: ToggleSize;
+  disabled?: boolean;
+  className?: string;
+  asChild?: boolean;
+  value?: string;
+}
+
+const sizes: Record<ToggleSize, { toggle: string; thumb: string }> = {
+  medium: {
+    toggle: "w-[30px] h-[20px]",
+    thumb: "w-[14px] h-[14px]",
+  },
+  large: {
+    toggle: "w-9 h-6",
+    thumb: "w-[18px] h-[18px]",
+  },
+};
+
+const Toggle: React.FC<ToggleProps> = ({
+  label,
+  defaultChecked = false,
+  checked,
+  onCheckedChange,
+  size = "medium",
+  disabled = false,
+  className = "",
+  asChild = false,
+  value,
+}) => {
+  const [internalChecked, setInternalChecked] = React.useState(defaultChecked);
+
+  React.useEffect(() => {
+    setInternalChecked(defaultChecked);
+  }, [defaultChecked]);
```

Comment:
not sure the need for `useEffect` here, because defaultChecked is only listened once (which is at the start of render). 

And you've already set that in `internalChecked` useState

Reply:
oh. actually what im trying to do is to synchronize controlled and uncontrolled states, and handle prop changes.
The component can work in both controlled and uncontrolled modes.
When used as a controlled component (i.e., checked prop is provided), this effect ensures that the internal state (internalChecked) stays in sync with the external controlled state (controlledChecked).

updated better implementation on next PR